assets/src/setup/components/unsaved-changes-warning/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useContext, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Options } from '../options-context-provider';
+
+export function WizardUnsavedChangesWarning() {
+	const { hasChanges, hasSaved } = useContext( Options );
+
+	/**
+	 * Warns the user if there are unsaved changes before leaving the wizard.
+	 *
+	 * @param {Event} event `beforeunload` event.
+	 */
+	const warnIfUnsavedChanges = useCallback( ( event ) => {
+		if ( hasChanges && ! hasSaved ) {
+			event.returnValue = __( 'This page has unsaved changes. Are you sure you want to leave?', 'amp' );
+		}
+
+		return null;
+	}, [ hasChanges, hasSaved ] );
+
+	useEffect( () => {
+		window.addEventListener( 'beforeunload', warnIfUnsavedChanges );
+
+		return () => {
+			window.removeEventListener( 'beforeunload', warnIfUnsavedChanges );
+		};
+	}, [ warnIfUnsavedChanges ] );
+
+	return null;
```

Comment:
I'm getting a warning here:

![image](https://user-images.githubusercontent.com/134745/83555681-837ff300-a4c3-11ea-8b37-12600280a2df.png)

Reply:
I'm not seeing this one -- is this coming from the ESLint config in the project? It looks like it might be misinterpreting the function as a class.

Reply:
You resolved the issue by adding an explicit `@return` declaration.